CONTRIBUTING.md

```diff
@@ -1,12 +1,80 @@
 # Contributing
 
-One of the easiest ways for you to contribute is to participate in discussions on GitHub issues. Right now, pull requests are not being accepted. The team has plans to open up the repository for contributions soon and will have a list of areas that will be up for grabs/need help. Check back in the coming weeks!
+One of the easiest ways for you to contribute is to participate in discussions on GitHub issues. You can also contribute by submitting pull requests with code changes.
 
 ## General feedback and discussions?
 Start a discussion on the [repository issue tracker](https://github.com/dotnet/upgrade-assistant/issues).
 
 ## Bugs and feature requests?
 For non-security related bugs, please [log a new issue](https://github.com/dotnet/upgrade-assistant/issues).
 
+## How to submit a PR 
+
+We are always happy to see PRs from community members both for bug fixes as well as new features.
+To help you be successful we've put together a few simple rules to follow when you prepare to contribute to our codebase:
+
+**Finding an issue to work on**
+
+  We've created a separate bucket of issues, which would be great candidates for community members to contribute to. We mark these issues with `help wanted` label. You can find all these issues [here](https://github.com/dotnet/upgrade-assistant/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22).
+  
+Within that set, we have additionally marked issues which are good candidates for first-time contributors. Those do not require too much familiarity with the codebase and are more novice-friendly. Those are marked with `good first issue` label. The full list of such issues can be found [here](https://github.com/dotnet/upgrade-assistant/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22).
+  
+If there is some other area not included here where you want to contribute to, first open an issue to describe the problem you are trying to solve and state that you're willing to contribute a fix for it. We will then discuss in that issue, if this is an issue we would like addressed and the best approach to solve it.
+  
+**Before writing code**
+
+  This can save you a lot of time. We've seen PRs, where customers would solve an issue in a way, which either wouldn't fit into upgrade-assistant because of how it's designed or it would change upgrade-assistant in a way, which is not something we'd like to do. To avoid these situations, we encourage customers to discuss the preferred design with the team first. To do so, file a new `design proposal` issue, link to the issue you'd like to address and provide detailed information about how you'd like to solve a specific problem. We triage issues peridocially and it will not take long for a team member to engage with you on that proposal.
```

Comment:
```suggestion
  This can save you a lot of time. We've seen PRs where customers solve an issue in a way which either wouldn't fit into upgrade-assistant because of how it's designed or would change upgrade-assistant in a way which is not something we'd like to do. To avoid these situations, we encourage customers to discuss the preferred design with the team first. To do so, file a new `design proposal` issue, link to the issue you'd like to address and provide detailed information about how you'd like to solve a specific problem. We triage issues periodically and it will not take long for a team member to engage with you on that proposal.
```